plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/LinuxExec.kt

```diff
@@ -28,18 +28,18 @@ open class LinuxExec @Inject constructor(objects: ObjectFactory) : ConclaveTask(
     val tag: Property<String> = objects.property(String::class.java)
 
     @get:Input
-    val tagLatest: Property<String> = objects.property(String::class.java)
+    val buildInDocker: Property<Boolean> = objects.property(Boolean::class.java)
 
     @get:Input
-    val buildInDocker: Property<Boolean> = objects.property(Boolean::class.java)
+    val useInternalDockerRegistry: Property<Boolean> = objects.property(Boolean::class.java)
 
     override fun action() {
         // This task should be set as a dependency of any task that requires executing a command in the context
         // of a Linux system or container.
         // Building the enclave requires docker container to make the experience consistent between all OSs.
         // This helps with using Gramine too, as it's included in the docker container and users don't need to
         // installed it by themselves. Only Python Gramine enclaves are built outside the container.
-        if (buildInDocker(buildInDocker)) {
+        if (buildInDocker(buildInDocker) && !useInternalDockerRegistry.get()) {
```

Comment:
On line 56 we threw an exception with a message to notify the user that they need to install docker. I presume we don't get that same exception now when they run without building the image? Can you try testing your changes with the flag set to true but docker not installed. The user should get the same message telling them to install docker.